### PR TITLE
feat: add typography component from XDS to archetype

### PR DESCRIPTION
**Pull request by CachedaCodes**
[EX-7616](https://searchbroker.atlassian.net/browse/EX-7616)

## Motivation and context
Applying the new XDS component of Typography to the Archetype.
The required changes are in the Task links. This [confluence table](https://searchbroker.atlassian.net/wiki/spaces/~168436735/pages/4077846529/Typography+component+in+Archettyype) summarizes what's needed

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

### CODE REVIEW

**Comment by tiborux** [comment]
You have to disable some components: 

```
/** @type {import('@empathyco/x-tailwindcss').PluginOptions} */
module.exports = {
  components() {
    return {
      '.suggestion': undefined,
      '.input': undefined
    };
  },
};
```

these components are still not used, we should wait for its task.

File: src/components/desktop/desktop-toolbar.vue [hunk 1]
```diff
@@ -6,7 +6,8 @@
   >
     <span
       v-html="$t('totalResults.message', { totalResults: $x.totalResults, query })"
-      class="x-list__item--expand x-font-size--05"
+      class="x-text1 x-list__item--expand"
```
Comment (tiborux) [inline src/components/desktop/desktop-toolbar.vue:9]: Ypu have to change it in mobile too:

image

It has to be md in mobile

image

File: src/components/search/spellcheck-message.vue [hunk 2]
```diff
@@ -1,7 +1,11 @@
 <template>
   <Spellcheck v-if="$x.totalResults > 0" #default="{ query }" class="x-message">
     <p>
-      <span v-html="$t('spellcheck.message', { query })" class="x-text" />
+      <span
```
Comment (tiborux) [inline src/components/search/spellcheck-message.vue:4]: we should fix this:

image

Comment (tiborux) [reply src/components/search/spellcheck-message.vue:4]: the button is not align and it doesn't have the right size?

File: src/i18n/messages/fr.messages.json [hunk 3]
```diff
@@ -93,10 +93,10 @@
       "addToCart": "AJOUTER AU PANIER"
     },
     "spellcheck": {
-      "message": "Aucun résultat trouvé pour <span class=\"x-text x-text--accent x-text--bold\">'{query}'</span>. Nous vous montrons les résultats pour"
+      "message": "Aucun résultat trouvé pour <span class=\"x-font-bold  x-text-auxiliary-50\">'{query}'</span>. Nous vous montrons les résultats pour"
```
Comment (tiborux) [inline src/i18n/messages/fr.messages.json:96]: should be a title3?

Comment (tiborux) [inline src/i18n/messages/fr.messages.json:96]:
```suggestion
      "message": "Aucun résultat trouvé pour <span class=\"x-title3 x-text-auxiliary-50\">'{query}'</span>. Nous vous montrons les résultats pour"
```

File: src/i18n/messages/pt.messages.json [hunk 4]
```diff
@@ -90,10 +90,10 @@
       "addToCart": "ADICIONAR AO CARRINHO"
     },
     "spellcheck": {
-      "message": "Sem resultados encontrados para <span class=\"x-text--accent\">'{query}'</span>. Aqui estão os resultados para"
+      "message": "Sem resultados encontrados para <span class=\"x-font-bold  x-text-auxiliary-50\">'{query}'</span>. Aqui estão os resultados para"
```
Comment (tiborux) [inline src/i18n/messages/pt.messages.json:93]:
```suggestion
      "message": "Sem resultados encontrados para <span class=\"x-title3 x-text-auxiliary-50\">'{query}'</span>. Aqui estão os resultados para"
```

File: src/i18n/messages/pt.messages.json [hunk 5]
```diff
@@ -90,10 +90,10 @@
...
     },
     "noResults": {
-      "message": "<span>Não encontramos resultados para<span class=\"x-text x-font-color--neutral-10 x-text--bold x-font-size--05\">\"{query}\"</span></span> Talvez você pode se interessar por esses produtos"
+      "message": "<span>Não encontramos resultados para<span class=\"x-font-bold\">\"{query}\"</span></span> Talvez você pode se interessar por esses produtos"
```
Comment (tiborux) [inline src/i18n/messages/pt.messages.json:96]:
```suggestion
      "message": "<span>Não encontramos resultados para<span class=\"x-title3\">\"{query}\"</span></span> Talvez você pode se interessar por esses produtos"
```

File: src/i18n/messages/fr.messages.json [hunk 6]
```diff
@@ -93,10 +93,10 @@
...
     },
     "noResults": {
-      "message": "<span>Désolé mais il n'y a pas de résultats pour <span class=\"x-text x-font-color--neutral-10 x-text--bold x-font-size--05\">\"{query}\"</span></span>Vous pourriez être intéressé par ces produits"
+      "message": "<span>Désolé mais il n'y a pas de résultats pour <span class=\"x-font-bold\">\"{query}\"</span></span>Vous pourriez être intéressé par ces produits"
```
Comment (tiborux) [inline src/i18n/messages/fr.messages.json:99]:
```suggestion
      "message": "<span>Désolé mais il n'y a pas de résultats pour <span class=\"x-title3\">\"{query}\"</span></span>Vous pourriez être intéressé par ces produits"
```

File: src/i18n/messages/en.messages.json [hunk 7]
```diff
@@ -93,10 +93,10 @@
       "addToCart": "ADD TO CART"
     },
     "spellcheck": {
-      "message": "No results found for <span class=\"x-text x-text--accent x-text--bold\">'{query}'</span>. We show you results for"
+      "message": "No results found for <span class=\"x-font-bold x-text-auxiliary-50\">'{query}'</span>. We show you results for"
```
Comment (tiborux) [inline src/i18n/messages/en.messages.json:96]:
```suggestion
      "message": "No results found for <span class=\"x-title3 x-text-auxiliary-50\">'{query}'</span>. We show you results for"
```

File: src/i18n/messages/en.messages.json [hunk 8]
```diff
@@ -93,10 +93,10 @@
...
     },
     "noResults": {
-      "message": "<span>Sorry but there are no results for <span class=\"x-text x-font-color--neutral-10 x-text--bold x-font-size--05\">\"{query}\"</span></span>You may be interested in these products"
+      "message": "<span>Sorry but there are no results for <span class=\"x-font-bold\">\"{query}\"</span></span>You may be interested in these products"
```
Comment (tiborux) [inline src/i18n/messages/en.messages.json:99]:
```suggestion
      "message": "<span>Sorry but there are no results for <span class=\"x-title3\">\"{query}\"</span></span>You may be interested in these products"
```

File: src/i18n/messages/es.messages.json [hunk 9]
```diff
@@ -90,10 +90,10 @@
       "addToCart": "AÑADIR AL CARRITO"
     },
     "spellcheck": {
-      "message": "No se han encontrado resultados para <span class=\"x-text--accent\">'{query}'</span>. Aquí tienes resultados para"
+      "message": "No se han encontrado resultados para <span class=\"x-font-bold x-text-auxiliary-50\">'{query}'</span>. Aquí tienes resultados para"
```
Comment (tiborux) [inline src/i18n/messages/es.messages.json:93]:
```suggestion
      "message": "No se han encontrado resultados para <span class=\"x-title3 x-text-auxiliary-50\">'{query}'</span>. Aquí tienes resultados para"
```

File: src/i18n/messages/es.messages.json [hunk 10]
```diff
@@ -90,10 +90,10 @@
...
     },
     "noResults": {
-      "message": "<span>No se han encontrado resultados para <span class=\"x-text x-font-color--neutral-10 x-text--bold x-font-size--05\">\"{query}\"</span></span>Quizás te interesen estos productos"
+      "message": "<span>No se han encontrado resultados para <span class=\"x-font-bold\">\"{query}\"</span></span>Quizás te interesen estos productos"
```
Comment (tiborux) [inline src/i18n/messages/es.messages.json:96]:
```suggestion
      "message": "<span>No se han encontrado resultados para <span class=\"x-title3\">\"{query}\"</span></span>Quizás te interesen estos productos"
```

File: src/components/mobile/mobile-next-query-preview.vue [hunk 11]
```diff
@@ -9,21 +9,21 @@
     "
     :suggestion="nextQuery"
   >
-    <i18n class="x-title3 x-next-query-preview__title" tag="span" path="nextQueryPreview.message">
+    <i18n class="x-text1 x-next-query-preview__title" tag="span" path="nextQueryPreview.message">
```
Comment (tiborux) [inline src/components/mobile/mobile-next-query-preview.vue:12]: I think this should be title3

```suggestion
    <i18n class="x-title3 x-next-query-preview__title" tag="span" path="nextQueryPreview.message">
```

File: src/components/search/no-results-message.vue [hunk 12]
```diff
@@ -2,8 +2,8 @@
   <span
     v-if="$x.noResults"
     v-html="$t('noResults.message', { query: $x.query.search })"
-    class="x-no-results-message x-message x-text"
-    :class="{ 'x-font-size--05': $x.device === 'desktop' }"
+    class="x-text1 x-no-results-message x-message"
+    :class="{ 'x-text1-lg': $x.device === 'desktop' }"
```
Comment (javieri-empathy) [inline src/components/search/no-results-message.vue:6]: question: Wasn't using tailwind responsive prefix one of the reasons to create the XDS using tailwind? Why do we still use dynamic classes?
```suggestion
    class="desktop:x-text1-lg"
```

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

**Comment by javieri-empathy** [comment]
Also, when are we going to deprecate the old styles?

File: src/components/search/spellcheck-message.vue (hunk 2 above)
Comment (CachedaCodes) [reply src/components/search/spellcheck-message.vue:4]: I already shared this with the design team. We'll probably have to see how to fix this with the button component.

File: src/i18n/messages/fr.messages.json (hunk 3 above)
Comment (CachedaCodes) [reply src/i18n/messages/fr.messages.json:96]: I'm treating the spellcheck component as something like a message component and using the styles for that one. I'm asking design anyway for confirmation

image

Comment (CachedaCodes) [reply src/i18n/messages/fr.messages.json:96]: I'm treating it as a message component, the same as the no results one, and it's `text-1`

image

File: src/i18n/messages/en.messages.json (hunk 8 above)
Comment (CachedaCodes) [reply src/i18n/messages/en.messages.json:99]: I'm seeing `title-1` in the table for the no results message component

File: src/components/mobile/mobile-next-query-preview.vue (hunk 11 above)
Comment (CachedaCodes) [reply src/components/mobile/mobile-next-query-preview.vue:12]: I think it should be `x-text1-lg`

File: package.json [hunk 13]
```diff
@@ -43,7 +43,7 @@
   "devDependencies": {
     "@cypress/webpack-preprocessor": "~5.11.1",
     "@empathyco/eslint-plugin-x": "^2.0.0-alpha.15",
-    "@empathyco/x-tailwindcss": "^0.2.0-alpha.28",
+    "@empathyco/x-tailwindcss": "^0.2.0-alpha.35",
```
[the remark this answers is not in the document]
Comment (CachedaCodes) [reply package.json:46]: I decided not to change them unless they were in elements using the new XDS classes, expecting that there'll be tasks to replace old utility classes to XDS ones. If you think it's better to do it here I'll change them.

File: src/components/search/no-results-message.vue (hunk 12 above)
Comment (CachedaCodes) [reply src/components/search/no-results-message.vue:6]: True, I'll search for more and change them.

**Comment by CachedaCodes** [comment]
> Also, when are we going to deprecate the old styles?

I'm guessing when all the new components are implemented in the XDS.

File: src/components/desktop/desktop-next-query-preview.vue [hunk 14]
```diff
@@ -4,20 +4,20 @@
...
     :suggestion="nextQuery"
   >
-    <i18n class="x-title3" tag="span" path="nextQueryPreview.message">
+    <i18n class="x-text1 x-text1-lg" tag="span" path="nextQueryPreview.message">
       <template #query>
-        <span class="x-text x-text--bold x-title3">{{ $x.query.nextQueries }}</span>
+        <span class="x-title3">{{ $x.query.nextQueries }}</span>
       </template>
     </i18n>
     <div class="x-list x-list--horizontal x-list--align-center x-list--gap-02">
```
[the remark this answers is not in the document]
Comment (CachedaCodes) [reply src/components/desktop/desktop-next-query-preview.vue:12]: I'm not seeing it, can you screenshot?

[the remark this answers is not in the document]
Comment (javieri-empathy) [reply src/components/desktop/desktop-next-query-preview.vue:12]: Correct (baseline): 
image

Wrong (center):
image

**Comment by javieri-empathy** [comment]
> > Also, when are we going to deprecate the old styles?
> 
> I'm guessing when all the new components are implemented in the XDS.

I thought that process was meant to be progressive. We already deprecated button styles for example.

File: package.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply package.json:46]: I'm not sure that makes much sense... it sounds like a task that will never be done. FMPOV it is better to replace them here and deprecate typography in the old DS ASAP.

File: src/components/search/no-results-message.vue [hunk 15]
```diff
@@ -19,7 +18,7 @@
     > span {
       width: 100%;
 
-      .x-text {
+      span {
```
Comment (javieri-empathy) [inline src/components/search/no-results-message.vue:21]: You can replace this style if we enable the `wordBreak` core plugin in the tailwind config and we add the `x-break-words` class: https://tailwindcss.com/docs/word-break

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: src/components/desktop/desktop-next-query-preview.vue [hunk 16]
```diff
@@ -4,20 +4,20 @@
     class="x-list x-background--neutral-95 x-list--gap-05 x-padding--06"
     :suggestion="nextQuery"
   >
-    <i18n class="x-title3" tag="span" path="nextQueryPreview.message">
+    <i18n class="x-text1 x-text1-lg" tag="span" path="nextQueryPreview.message">
       <template #query>
-        <span class="x-text x-text--bold x-title3">{{ $x.query.nextQueries }}</span>
+        <span class="x-title3">{{ $x.query.nextQueries }}</span>
```
[the remark this answers is not in the document]
Comment (CachedaCodes) [reply src/components/desktop/desktop-next-query-preview.vue:9]: Discussed this one with design. Decision is leaving it as it is for now and that they're aware.

File: src/components/desktop/desktop-next-query-preview.vue (hunk 14 above)
[the remark this answers is not in the document]
Comment (CachedaCodes) [reply src/components/desktop/desktop-next-query-preview.vue:12]: Changed except for query previews, it looks to me that in that component the alignment is right.

**Comment by CachedaCodes** [comment]
> 



> > > Also, when are we going to deprecate the old styles?
> > 
> > 
> > I'm guessing when all the new components are implemented in the XDS.
> 
> I thought that process was meant to be progressive. We already deprecated button styles for example.

I replaced every instance I've found of `x-font`.

File: src/components/desktop/desktop-toolbar.vue [hunk 17]
```diff
@@ -24,9 +24,7 @@
           x-background--accent
           x-border-radius--20
           x-padding--01 x-padding--right-03 x-padding--left-03
-          x-text--light
-          x-font-color--neutral-100
-          x-font-size--04
+          x-text-neutral-0 x-text-xs x-leading-[1.5]
```
Comment (CachedaCodes) [inline src/components/desktop/desktop-toolbar.vue:27]: This element will be converted to a XDS component. This just replaces the old DS classes with tailwind ones with the same rules

File: src/components/mobile/mobile-toolbar.vue [hunk 18]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div class="x-list x-list--horizontal x-list--align-center x-list__item--expand">
-    <span v-if="$x.totalResults" class="x-list__item--expand x-font-size--05">
+    <span v-if="$x.totalResults" class="x-title1 x-list__item--expand">
```
Comment (javieri-empathy) [inline src/components/mobile/mobile-toolbar.vue:3]: issue: I don't believe this is the intended style:
image

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: src/components/mobile/mobile-toolbar.vue (hunk 18 above)
Comment (CachedaCodes) [reply src/components/mobile/mobile-toolbar.vue:3]: Leftover test I believe

**Review by javieri-empathy — APPROVED** [review]

File: src/components/search/sort.vue [hunk 19]
```diff
@@ -18,7 +18,7 @@
         <template #default="{ item, isSelected }">
           <RadioButtonSelectedIcon v-if="isSelected" class="x-icon--l" />
           <RadioButtonUnselectedIcon v-else class="x-icon--l" />
-          <span class="x-font-color--neutral-10" :class="{ 'x-text--bold': isSelected }">
+          <span class="x-text-neutral-90" :class="{ 'x-text--bold': isSelected }">
```
Comment (ajperezbau) [inline src/components/search/sort.vue:21]: Should we remove that `x-text--bold` class?

File: src/components/results/recommendations.vue [hunk 20]
```diff
@@ -4,7 +4,7 @@
     class="x-list x-list--align-center x-list--padding-07 x-list--padding-bottom"
     :class="{ 'x-list--align-start': $x.device === 'mobile' }"
   >
-    <h1 class="x-title1" :class="{ 'x-padding--left-05': $x.device === 'mobile' }">
+    <h1 class="x-title1" :class="{ 'x-title1-sm x-padding--left-05': $x.device === 'mobile' }">
```
Comment (ajperezbau) [inline src/components/results/recommendations.vue:7]: What do you think about using here the tailwind device classes, to do it in the same way everywhere?
Something like:

```suggestion
    <h1 class="x-title1 x-title1-sm desktop:x-title1-md" :class="{ 'x-padding--left-05': $x.device === 'mobile' }">
```

**Review by ajperezbau — CHANGES_REQUESTED** [review]
Awesome 👌

File: src/components/search/sort.vue (hunk 19 above)
Comment (CachedaCodes) [reply src/components/search/sort.vue:21]: We should change it to `x-font-bold` yes

File: src/components/results/recommendations.vue (hunk 20 above)
Comment (CachedaCodes) [reply src/components/results/recommendations.vue:7]: We haven't discussed what to do about the mobile only classes so I would leave it as is for now

File: src/tailwind/plugin-options.js [hunk 21]
```diff
@@ -1,2 +1,14 @@
...
-module.exports = {};
+module.exports = {
+  components() {
+    return {
+      '.suggestion': undefined
+    };
+  },
+  theme: {
+    screens: {
+      tablet: '640px',
```
[the remark this answers is not in the document]
Comment (ajperezbau) [reply src/tailwind/plugin-options.js]: Task to have the value in one place.

https://searchbroker.atlassian.net/browse/EX-7715

**Review by ajperezbau — APPROVED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**